tket/src/WeightSubgrMono/include/WeightSubgrMono/Searching/WeightUpdater.hpp

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019-2022 Cambridge Quantum Computing
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include <map>
+#include <optional>
+#include <set>
+#include <string>
+
+#include "../GraphTheoretic/GeneralStructs.hpp"
+
+namespace tket {
+namespace WeightedSubgraphMonomorphism {
+
+class SearchNodeWrapper;
+struct FixedData;
+
+/** Some new pv->tv assignments have been made in a search node.
+ * Updates the total weight (scalar product) of this node,
+ * whenever new edges become assigned.
+ */
+class WeightUpdater {
```

Comment:
The whole point of many different classes is to make each individual class smaller and simpler, so I think it would be a backward step to merge the weight updater class with the fixed data class. Fixed data is just a handy single location to put all the different algorithms (encoded as classes).

Reply:
I understand the rationale - it may have been harder to track because these class objects were initialised at the construction of the fixed data class so at times it felt like these objects were appearing from nowhere. Are there cases where this separate class could be initialised, ran and destroyed at their point of use in the code as oppposed to initialised separately? that would ease understanding